Modules/System/Cryptography Management/src/CertificateRequestImpl.Codeunit.al

```diff
@@ -0,0 +1,153 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 1295 "CertificateRequest Impl."
+{
+    Access = Internal;
+
+    var
+        DotNetRSACryptoServiceProvider: DotNet RSACryptoServiceProvider;
+        DotNetCertificateRequest: DotNet CertificateRequest;
+
+    procedure InitializeRSA(KeySize: Integer; IncludePrivateParameters: Boolean; var KeyAsXmlString: Text)
```

Comment:
We could do it that way, but doesn't it deviate a bit from the pattern used for the other "classes"? And also, it is possible to use a ECDSA key instead of RSA (even if I have not implemented that here) which might make it a bit illogical?